zap/src/main/java/org/zaproxy/zap/extension/stdmenus/PopupMenuItemIncludeSiteInContext.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.stdmenus;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.zap.model.Context;
+import org.zaproxy.zap.view.popup.PopupMenuItemIncludeInContext;
+
+class PopupMenuItemIncludeSiteInContext extends PopupMenuItemIncludeInContext {
+
+    private static final long serialVersionUID = 1L;
+    private static Logger log = LogManager.getLogger(PopupMenuItemIncludeSiteInContext.class);
+
+    PopupMenuItemIncludeSiteInContext() {
+        super();
+    }
+
+    PopupMenuItemIncludeSiteInContext(Context context) {
+        super(context);
+    }
+
+    @Override
+    public String getParentMenuName() {
+        return Constant.messages.getString("context.includesite.popup");
+    }
+
+    @Override
+    protected String normaliseUrl(String url) {
+        try {
+            URI urlWithPath = new URI(url, true);
+            return urlWithPath.getScheme() + "://" + urlWithPath.getHost() + ".*";
```

Comment:
Doesn’t this need to getAuthority in order to include port if specified?

I guess that’d get caught by the trailing .* but I’m not sure if it’s safer to err on the side of generality or specificity.

Reply:
It does, and better add tests.